addons/mrp/models/mrp_production.py

```diff
@@ -1379,9 +1379,10 @@ def _post_inventory(self, cancel_backorder=False):
     def _get_name_backorder(self, name, sequence):
         if not sequence:
             return name
-        seq_back = "-" + "0" * (SIZE_BACK_ORDER_NUMERING - 1 - int(math.log10(sequence))) + str(sequence)
-        if re.search("-\\d{%d}$" % SIZE_BACK_ORDER_NUMERING, name):
-            return name[:-SIZE_BACK_ORDER_NUMERING-1] + seq_back
+        num_size = int(self.env['ir.config_parameter'].get_param('mrp_production.SIZE_BACK_ORDER_NUMERING', SIZE_BACK_ORDER_NUMERING))
+        seq_back = "-" + "0" * (num_size - 1 - int(math.log10(sequence))) + str(sequence)
+        if re.search("-\\d{%d}$" % num_size, name):
```

Comment:
A other solution is to fix the regex (my bad) to take in account a variable size:
```python
if re.search(r"-\d+$" % num_size, name):
```
Then when someone exceed 1000, it will manage automatically:
- MO/00001-001
- ...
- MO/00001-999
- MO/00001-1000
- MO/00001-1001

It avoids the System Parameters (then avoid manual fix in all server with this issue - also doesn't need to recreate a new MO to fix the issue), but the name of MO can have the different length name (it is a issue?)

What do you think ?

Reply:
OK for me, that's even better!

Reply:
Done! Check the new version